docs/zedctr.md

```diff
@@ -0,0 +1,96 @@
+
+* `libfdt` via `apk` for `arm64` only
+* symlink `/usr/lib/libpcap.so` to `libpcap.so.1`
+* generate a host ssh key, configure `/etc/ssh/sshd_config` to permit root login via ssh, and change the ssh root key. It is unclear why we are doing this for several reasons:
+    * We are generating the root password at package build time, storing secrets in a static OCI image. 
+    * We are generating the ssh host key at package build time, causing all hosts to have the same ssh host key.
+    * We are allowing root login via ssh
+* copy the local [pkg/zedctr/rootfs](../pkg/zedctr/rootfs/) into the image at `/`
+* Set the `CMD /init.sh`
+
+#### lisp-install
+
+lisp files are copied as follows:
+
+```
+COPY --from=lisp /lisp/lisp-ztr /opt/zededa/bin/
+COPY --from=lisp /lisp /opt/zededa/lisp/
+COPY --from=lisp /usr/bin/pydoc /usr/bin/smtpd.py /usr/bin/python* /usr/bin/
+COPY --from=lisp /usr/lib/libpython* /usr/lib/libffi.so* /usr/lib/
+COPY --from=lisp /usr/lib/python2.7 /usr/lib/python2.7/
+```
+
+#### zedctr/rootfs
+
+[zedctr/rootfs](../pkg/zedctr/rootfs) is a small group of files that are installed in the root to `/` of the `zedctr` container. This document will not list every file, but will give the general purposes.
+
+* [blink.sh](../pkg/zedctr/rootfs/blink.sh) - appears to be a simple utility to exercise `/dev/sda` by copying 22 blocks of 4M to `/dev/null`, a process that should take about 200ms, then sleeps 200ms, repeating 3 times, then sleeping 1200ms, then repeating. Its usage is unknown.
+* [sos.sh](../pkg/zedctr/rootfs/sos.sh) - similar to blink, but an SOS pattern instead. Its usage is unknown.
+* [dhcpcd.conf](../pkg/zedctr/rootfs/dhcpcd.conf) - dhcp client daemon config file. Unclear why it is in `/` and not under `/etc`.
+* [init.sh](../pkg/zedctr/rootfs/init.sh) - entrypoint command for the `zedctr` container. Starts with a comment, "This *really* needs to be replaced with tini+monit ASAP." It functions as a process launcher. It does the following:
+    * start sshd
+    * start crond for logrotate
+    * disable hardware TCP offloading
+    * set the nameserver to `8.8.8.8`
+    * start Xen with logging defined
+    * runs optional `/opt/zededa/bin/device-steps.sh`
+    * follow tails (`tail -f`) all files in `/var/log/*.log` as well as `/dev/null`, apparently to ensure it never exits, terminating the container.
+* [etc/](../pkg/zedctr/rootfs/etc/) - various config files for `logrotate` and `nsswitch`
+* [config/](../pkg/zedctr/rootfs/config) - empty directory, required for runtime
+
+### Modularization Path
+
+This should be modularized, with each one its own package and service in the runtime. Each of the listed services should be its own container, managed by containerd (as is normal when listed in linuxkit `services:`). Each should also have its own minimum required capabilities, and should avoid the host PID and networking namespaces, where possible.
```

Comment:
We want the different microservices to run in separate service domains, not just in separate containers. It would make sense to mention that as part of the modularization path.
(That is most important for the services in go-provision; those which are started by device-steps.sh)

Reply:
When you say, "separate service domains", you mean separate domU for each? Do we have that now? Or is that just a plan? Why do they need to be in separate virtualized hardware spaces with their own kernels vs separate virtualized OS (i.e. container)?

Reply:
@deitch it is just a plan. That's why I was talking to you about qrexec by the way -- once we go full DomU separation it will be inevitable

Reply:
OK, I captured it.